Will_and_Pete/Assets/Scripts/Player/GunView.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using UnityEngine;
+
+namespace Assets.Scripts.Player
+{
+    public class GunView : MonoBehaviour
+    {
+        [SerializeField] private Transform gunTurnAxis;
+        [SerializeField] private ShootingSettings settings;
+        [SerializeField] private SpriteRenderer spRend;
+        [SerializeField] private GameObject gunFireAnimation;
+        [SerializeField] private Transform gunNozzlePosition;
+        private float currentLineFadeTime;
+        private LineRenderer lineRenderer;
+
+        private void Awake()
+        {
+            lineRenderer = GetComponent<LineRenderer>();
+        }
+
+        private void Update()
+        {
+            FadeFireLine();
+        }
+
+        public void DrawFireLine(Vector2 endPos)
+        {
+            Vector3[] linePositions = { gunNozzlePosition.position, (Vector3)endPos };
+            lineRenderer.SetPositions(linePositions);
+            currentLineFadeTime = 0;
+            var temp = Instantiate(gunFireAnimation);
+            temp.transform.localPosition = gunNozzlePosition.position;
+            temp.transform.rotation = gunTurnAxis.rotation;
+            temp.SetActive(true);
```

Comment:
spawn one at game start and use via active inactive. 
Not too urgent here but worth sticking too.